src/Validator.php

```diff
@@ -13,73 +13,65 @@
+ * @phpstan-type ValitronRule array<int|'message', ValitronRuleType|int|string|string[]|ValidatorCallback>
  */
 class Validator
 {
     use WarnDynamicPropertyTrait;
+
     /**
-     * Array of rules in following format which is natively supported by Valitron mapFieldsRules():
-     *  [
-     *      'foo' => [
-     *          ['required'],
-     *          ['integer', 'message'=>'test 1'],
-     *      ],
-     *      'bar' => [
-     *          ['email'],
-     *          ['lengthBetween', 4, 10, 'message'=>'test 2'],
-     *      ],
-     *  ];.
-     *
-     * @var array<string, array<string|array<string|int|\Closure(string, mixed, list<mixed>, list<mixed>): bool>>>
+     * @var list<ValidatorRule>
      */
     public array $rules = [];
 
+    public function __construct(Model $model)
+    {
+        $model->onHook(Model::HOOK_VALIDATE, \Closure::fromCallable([$this, 'validate']));
+    }
+
     /**
-     * Array of conditional rules in following format:
-     *  [
-     *      [
-     *          $conditions, // array of conditions
-     *          $then_rules, // array in $this->rules format which will be used if conditions are met
-     *          $else_rules  // array in $this->rules format which will be used if conditions are not met
-     *      ],
-     *  ].
+     * Set rules of particular field.
+     *
+     * @param array<string|ValitronRule>                      $rules
+     * @param array<string, mixed>|list<array<string, mixed>> $conditions
      *
-     * @var list<
-     *          array{
-     *              array<string, string>,
-     *              array<string, array<string|array<string|int|\Closure(string, mixed, list<mixed>, list<mixed>): bool>>>,
-     *              array<string, array<string|array<string|int|\Closure(string, mixed, list<mixed>, list<mixed>): bool>>>
-     *          }
-     *  >
+     * @return $this
      */
-    public array $if_rules = [];
-
-    public function __construct(Model $model)
+    public function rule(string $field, array $rules, ?string $activateOn = null, array $conditions = []): self
     {
-        $model->onHook(Model::HOOK_VALIDATE, \Closure::fromCallable([$this, 'validate']));
+        foreach ($rules as $rule) {
+            $validatorRule = new ValidatorRule($field, $rule);
+            if ($activateOn !== null) {
+                $validatorRule->setActivateOnResult($activateOn, $conditions);
+            }
+            $this->addValidatorRule($validatorRule);
+        }
+
+        return $this;
     }
 
     /**
      * Set one rule.
      *
-     * @param array<string|array<string|int|\Closure(string, mixed, list<mixed>, list<mixed>): bool>> $rules
-     *
      * @return $this
      */
-    public function rule(string $field, array $rules): self
+    public function addValidatorRule(ValidatorRule $validatorRule): self
```

Comment:
```suggestion
    public function addRule($rule): self
```

for future rule() -> addRule() refactor